stack/stack.go

```diff
@@ -40,26 +52,47 @@ type (
 	}
 )
 
-func (s S) Name() string {
-	if s.block.Name != "" {
-		return s.block.Name
+// New creates a new stack from configuration cfg.
+func New(root string, cfg hcl.Config) S {
+	name := cfg.Stack.Name
+	if name == "" {
+		name = filepath.Base(cfg.AbsDir())
+	}
+
+	return S{
+		name:        name,
+		Dir:         project.RelPath(root, cfg.AbsDir()),
+		description: cfg.Stack.Description,
+		after:       cfg.Stack.After,
+		before:      cfg.Stack.Before,
 	}
-	return filepath.Base(s.Dir)
 }
 
-func (s S) Description() string {
-	return s.block.Description
+// Name of the stack.
+func (s S) Name() string {
+	if s.name != "" {
+		return s.name
+	}
+	return s.Dir
 }
 
-func (s S) After() []string  { return s.block.After }
-func (s S) Before() []string { return s.block.Before }
+// Description of stack.
+func (s S) Description() string { return s.description }
+
+// After specifies the list of stacks that must run before this stack.
+func (s S) After() []string { return s.after }
 
-func (s S) IsChanged() bool    { return s.changed }
+// Before specifies the list of stacks that must run after this stack.
+func (s S) Before() []string { return s.before }
+
+// IsChanged tells if the stack is marked as changed.
+func (s S) IsChanged() bool { return s.changed }
+
+// SetChanged sets the changed flag of the stack.
 func (s *S) SetChanged(b bool) { s.changed = b }
 
-func (s S) String() string {
-	return s.Name()
-}
+// String representation of the stack.
+func (s S) String() string { return s.Name() }
```

Comment:
hmm maybe add more info to the string representation ? Would be good to know if it is changed when printing a stack, maybe :thinking: (I can just print with +v/#v too...so not sure, just wondering).

Reply:
not sure either, but if we format this here it's going to show everywhere. We have to talk about this I guess.